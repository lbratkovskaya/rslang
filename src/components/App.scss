$color-main-bg: #fafafa;
$color-main-shadow: #16181b;

main {
  //min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  z-index: 1;
  background: $color-main-bg;
  box-shadow: 0 12px 12px $color-main-shadow;
}

.page-exit {
  display: none;
}

.page-enter {
  opacity: 0.2;
}

.page-enter-active {
  opacity: 1;
  transition: all 600ms ease-out;
}