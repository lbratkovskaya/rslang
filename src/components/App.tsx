import React from 'react';

import { HashRouter, Switch, Route } from 'react-router-dom';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import AudioCallingPage from './AudioCallingPage';
import WordBook from './WordBook';
import MainPage from './MainPage';
import GameSavannah from './GameSavannah';
import GamesPage from './GamesPage';
import MemoryGame from './MemoryGame';
import GameSprint from './GameSprint';
import RandomPage from './RandomPage';
import Settings from './Settings';
import SignInPage from './SignInPage';
import SignUpPage from './SignUpPage';
import Statistics from './Statistics/Statistics';
import UserDictionary from './UserDictionary';
import { enterTimeout, exitTimeout, ROUTES } from '../constants';
import './App.scss';

const App: React.FC = () => {
  const timeout = { enter: enterTimeout, exit: exitTimeout };

  return (
    <HashRouter>
      <Route
        render={(routeProps) => (
          <TransitionGroup component={null}>
            <CSSTransition
              key={routeProps.location.pathname}
              timeout={timeout}
              classNames="page"
              mountOnEnter
              unmountOnExit>
              <Switch location={routeProps.location}>
                <Route exact path={ROUTES.root} component={MainPage} />
                <Route path={ROUTES.study} component={RandomPage} />
                <Route exact path={ROUTES.games.root} component={GamesPage} />
                <Route path={ROUTES.settings} component={Settings} />
                <Route path={ROUTES.wordBook.root} component={WordBook} />
                <Route path={ROUTES.dictionary} component={UserDictionary} />
                <Route path={ROUTES.games.memory} component={MemoryGame} />
                <Route path={ROUTES.games.savannah} component={GameSavannah} />
                <Route path={ROUTES.games.audio} component={AudioCallingPage} />
                <Route path={ROUTES.games.sprint} component={GameSprint} />
                <Route path={ROUTES.signIn} component={SignInPage} />
                <Route path={ROUTES.signUp} component={SignUpPage} />
                <Route path={ROUTES.statistics} component={Statistics} />
              </Switch>
            </CSSTransition>
          </TransitionGroup>
        )}
      />
    </HashRouter>
  );
};

export default App;
