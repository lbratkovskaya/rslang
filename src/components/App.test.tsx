import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import App from './App';

describe('<App /> testing', () => {
  const wrapper: ShallowWrapper<typeof App> = shallow(<App />);

  it('Must have a correct snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('Should use HashRouter', () => {
    expect(wrapper.find('HashRouter')).toHaveLength(1);
  });
});
