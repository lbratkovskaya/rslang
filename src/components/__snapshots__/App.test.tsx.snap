// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> testing Must have a correct snapshot 1`] = `
"<HashRouter>
  <Route render={[Function: render]} />
</HashRouter>"
`;
